content/blog/2018-09-04-birds-taxo-traits.md

```diff
@@ -0,0 +1,559 @@
+                             format_out = "sf_polygon")
+crs <- sf::st_crs(landkreis_konstanz)
+
+# get and filter data
+f_out_ebd <- "ebird/ebd_lk_konstanz.txt"
+
+library("magrittr")
+
+ebd <- auk::read_ebd(f_out_ebd) %>%
+  sf::st_as_sf(coords = c("longitude", "latitude"), 
+                crs = crs) 
+
+in_indices <- sf::st_within(ebd, landkreis_konstanz)
+
+ebd <- dplyr::filter(ebd, lengths(in_indices) > 0)
+
+ebd <- as.data.frame(ebd)
+
+ebd <- dplyr::filter(ebd, approved, lubridate::year(observation_date) > 2010)
+
+nrow(ebd)
+```
+
+    ## [1] 8599
+
+We’ll also need these two data.frames later: abundance by species, and
+dictionary of names.
+
+``` r
+abundance <- dplyr::count(ebd, scientific_name)
+
+dico <- unique(dplyr::select(ebd, scientific_name,
+                             common_name))
+```
+
+### Getting taxonomic information
+
+In this section I would like to get an idea of how much of the bird
+diversity is represented in the County of Constance. I want to draw a
+phylogenetic tree of the local species, and for that, I’ll first
+retrieve the classification for each species from
+[NCBI](https://www.ncbi.nlm.nih.gov/taxonomy) using the [`taxize`
+package](https://github.com/ropensci/taxize) that “allows users to
+search over many taxonomic data sources for species names (scientific
+and common) and download up and downstream taxonomic hierarchical
+information - among other things.”.
+
+We first query *uid*’s and then use the `classification` function,
+instead of passing the species name directly to `classification`,
+because the IDs are unique whereas results for species names aren’t.
+Rate limiting is thankfully managed by the package itself so we users do
+not need to worry about that.
+
+``` r
+ids <- taxize::get_uid(unique(ebd$scientific_name))
+classif <- taxize::classification(ids)
+
+fs::dir_create("taxo")
+save(classif, file = file.path("taxo", "classif.RData"))
+```
+
+There are 211 species, we get 211 elements in the classification
+(`sum(lengths(classif)==3)`), that is a list of data.frames:
+
+``` r
+load(file.path("taxo", "classif.RData"))
+
+str(classif[[1]])
+```
+
+    ## 'data.frame':    31 obs. of  3 variables:
+    ##  $ name: chr  "cellular organisms" "Eukaryota" "Opisthokonta" "Metazoa" ...
+    ##  $ rank: chr  "no rank" "superkingdom" "no rank" "kingdom" ...
+    ##  $ id  : chr  "131567" "2759" "33154" "33208" ...
+
+Now, we’ll represent the whole taxonomy as a tree, using the handy
+`taxize::class2tree` function and the great [`ggtree` package by
+Guangchuang Yu](https://guangchuangyu.github.io/software/ggtree/).
+
+``` r
+tree <- taxize::class2tree(classif)
+```
+
+``` r
+library("ggplot2")
+ggtree::ggtree(tree$phylo) +
+    ggtree::geom_tiplab(aes(), size = 2, vjust=0.25) +
+  xlim(0, 150)
+```
+
+![labelled tree of all
+species](/img/blog-images/2018-09-04-birds-taxo-traits/firsttree-1.png)
+
+This tree is… unreadable. But at this point, it’s worth remembering that
```

Comment:
I love this: "This tree is… unreadable. "

Reply:
Honesty FTW :joy: